src/xrpld/ledger/detail/View.cpp

```diff
@@ -303,8 +314,36 @@ accountHolds(
             amount.negate();
         }
         amount.setIssuer(issuer);
+
+        // if it's a LPToken, also need to check if issuers of the asset pair
+        // has frozen holder's trustline
+        if (view.rules().enabled(fixFrozenLPTokenTransfer) &&
+            zeroIfFrozen == fhZERO_IF_FROZEN)
+        {
+            auto const sleIssuer = view.read(keylet::account(issuer));
+            if (!sleIssuer)
+            {
+                amount.clear(Issue{currency, issuer});
+            }
+            else if (sleIssuer->isFieldPresent(sfAMMID))
+            {
+                auto const sleAmm =
+                    view.read(keylet::amm((*sleIssuer)[sfAMMID]));
+
+                if (!sleAmm ||
+                    isLPTokenFrozen(
+                        view,
+                        account,
+                        (*sleAmm)[sfAsset].get<Issue>(),
+                        (*sleAmm)[sfAsset2].get<Issue>()))
+                {
+                    amount.clear(Issue{currency, issuer});
+                }
+            }
```

Comment:
note: this code would likely need to be refactored similar to [deepfreeze](https://github.com/XRPLF/rippled/pull/5227/files#diff-31c845b2ac1b7fc26729d6aa4c01c06412d476063dedf8175f9d22e75e69fd3c) cc @vvysokikh1